### PR TITLE
feat: update required/optional response model

### PR DESCRIPTION
**Pull request by vincentdelmundo**
# Description
most of the changes in soure code are removal of null checking
and changes in UT are mostly removal of unnecessary cases (null case) and adding of default values (since some variables are no longer optional/null)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/responses/ping/Trigger.kt [hunk 1]
```diff
@@ -7,14 +7,14 @@ import com.google.gson.annotations.SerializedName
  */
 internal data class Trigger(
     @SerializedName("type")
-    val type: Int? = null,
+    val type: Int,
 
     @SerializedName("eventType")
     val eventType: Int? = null,
```
Comment (imaginaris) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/responses/ping/Trigger.kt:13]: The specs say that it's required only for `type` 1 (Event) but at the same time specs say that `1` is always returned. So I think we can set it to non-optional

Comment (vincentdelmundo) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/data/responses/ping/Trigger.kt]: updated

**Review by ra-mdusman-gani — APPROVED** [review]
Looks good to me

**Review by imaginaris — APPROVED** [review]
👍

**Pull request merged**